src/Shared/TryParseMethodCache.cs

```diff
@@ -31,7 +34,18 @@ public static bool HasTryParseMethod(ParameterInfo parameter)
                     return EnumTryParseMethod.MakeGenericMethod(type);
                 }
 
-                var staticMethods = type.GetMethods(BindingFlags.Public | BindingFlags.Static);
+                if (TryGetDateTimeTryPareMethod(type, out var methodDateInfo))
+                {
+                    return methodDateInfo;
+                }
+
+                if (TryGetNumberStylesTryGetMethod(type, out var methodInfo, out var _))
```

Comment:
I'm trying to get it so we combine the finding and `Func<Expression, MethodCallExpression>` logic. I don't think we need a method that just get's the `MethodInfo` anymore because then you just end up redoing a lot of the logic to figure out what the parameters should be. Instead we should generate that func right here using the currently-discarded out var.